src/librustc_trait_selection/traits/error_reporting/mod.rs

```diff
@@ -286,6 +286,20 @@ impl<'a, 'tcx> InferCtxtExt<'tcx> for InferCtxt<'a, 'tcx> {
                             .starts_with("std::convert::From<");
                         let is_unsize =
                             { Some(trait_ref.def_id()) == self.tcx.lang_items().unsize_trait() };
+                        let is_fn_trait = [
+                            self.tcx.lang_items().fn_trait(),
+                            self.tcx.lang_items().fn_mut_trait(),
+                            self.tcx.lang_items().fn_once_trait(),
+                        ]
+                        .contains(&Some(trait_ref.def_id()));
+                        let is_safe_target_feature_fn =
```

Comment:
can we just make this so that *any* target-feature function doesn't implement `Fn` traits? If they are unsafe, then that's two reasons for them not to implement the traits. Seems simpler this way.

Reply:
Definitely simpler and the extra note might be helpful.